Sage/carbon
# fix(draggable): prevent crash when dragging item from one container to another

### Proposed behaviour

For both `Draggable` and `FlatTable` (when used in draggable mode), when there is more than one draggable container on the page, prevent an error when attempting to drag an item from one container to another.

Note we do not support dragging the items from one container to another, we are just preventing it causing an error which crashes the page.

### Current behaviour

As shown in https://github.com/Sage/carbon/issues/4857 when dragging an item from one draggable container to another, it crashes the page as soon as you hover over the second container.

### Checklist




#### QA


### Additional context



### Testing instructions


The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

FlatTable draggable: https://codesandbox.io/s/jovial-zeh-u5j5q8 

Draggable component: https://codesandbox.io/s/funny-mestorf-nzcpsz?file=/src/index.js

**DO NOT USE THE SANDBOX FROM THE ISSUE FOR TESTING THE FIX.** It incorrectly sets multiple elements with the same id, which is not valid HTML. Having said that, the crash will no longer occur on the issue sandbox, you will just see strange behaviour with item re-ordering when dragging.
